HPXMLtoOpenStudio/resources/electric_panel.rb

```diff
@@ -0,0 +1,251 @@
+        load_based_capacity_values = LoadValues.new
+        calculate_load_based(hpxml_bldg, electric_panel, load_based_capacity_values, service_feeders_load_calculation_type)
+
+        capacity_types << service_feeders_load_calculation_type
+        capacity_total_watts << load_based_capacity_values.Load_CapacityW.round(1)
+        capacity_total_amps << load_based_capacity_values.Load_CapacityA.round(1)
+        capacity_headroom_amps << load_based_capacity_values.Load_HeadRoomA.round(1)
+      elsif service_feeders_load_calculation_type.include?('Meter-Based')
+        meter_based_capacity_values = LoadValues.new
+        calculate_meter_based(hpxml_bldg, electric_panel, meter_based_capacity_values, service_feeders_load_calculation_type)
+
+        capacity_types << service_feeders_load_calculation_type
+        capacity_total_watts << meter_based_capacity_values.Load_CapacityW.round(1)
+        capacity_total_amps << meter_based_capacity_values.Load_CapacityA.round(1)
+        capacity_headroom_amps << meter_based_capacity_values.Load_HeadRoomA.round(1)
+      end
+    end
+    electric_panel.capacity_types = capacity_types
+    electric_panel.capacity_total_watts = capacity_total_watts
+    electric_panel.capacity_total_amps = capacity_total_amps
+    electric_panel.capacity_headroom_amps = capacity_headroom_amps
+
+    breaker_spaces_values = BreakerSpacesValues.new
+    calculate_breaker_spaces(electric_panel, breaker_spaces_values)
+
+    electric_panel.breaker_spaces_total = breaker_spaces_values.BreakerSpaces_Total
+    electric_panel.breaker_spaces_occupied = breaker_spaces_values.BreakerSpaces_Occupied
+    electric_panel.breaker_spaces_headroom = breaker_spaces_values.BreakerSpaces_HeadRoom
+  end
+
+  # Get the heating system attached to the given service feeder.
+  #
+  # @param hpxml_bldg [HPXML::Building] HPXML Building object representing an individual dwelling unit
+  # @param [HPXML::ServiceFeeder] Object that defines a single electric panel service feeder
+  # @return [HPXML::HeatingSystem] The heating system referenced by the panel load
+  def self.get_panel_load_heating_system(hpxml_bldg, service_feeder)
+    hpxml_bldg.heating_systems.each do |heating_system|
+      next if !service_feeder.component_idrefs.include?(heating_system.id)
+
+      return heating_system
+    end
+    return
+  end
+
+  # Get the heat pump attached to the given service feeder.
+  #
+  # @param hpxml_bldg [HPXML::Building] HPXML Building object representing an individual dwelling unit
+  # @param [HPXML::ServiceFeeder] Object that defines a single electric panel service feeder
+  # @return [HPXML::HeatPump] The heat pump referenced by the panel load
+  def self.get_panel_load_heat_pump(hpxml_bldg, service_feeder)
+    hpxml_bldg.heat_pumps.each do |heat_pump|
+      next if !service_feeder.component_idrefs.include?(heat_pump.id)
+
+      return heat_pump
+    end
+    return
+  end
```

Comment:
Combine these methods into one, they are basically identical. Just pass in `hpxml_bldg.heating_systems` or `hpxml_bldg.heat_pumps`.